tests/test_action.py

```diff
@@ -29,11 +30,24 @@ def _get_website_config(cache: Sequence[S3ArtifactCustomMetadataConfig] = ()) ->
     )
 
 
-def _get_upload_commands(pattern: str, mime_type: str, max_age: str) -> tuple[str, str]:
+def _get_metadata_command():
+    return (
+        f"aws s3 cp {ARTIFACTS_BUCKET} {ARTIFACTS_BUCKET} --recursive --no-progress --exclude '*' --include '*.html' "
+        f"--metadata '{json.dumps({
+            "X-Frame-Options": "SAMEORIGIN",
+            "Content-Security-Policy": "frame-src 'self'; frame-ancestors 'self'; object-src 'none';",
+            "Strict-Transport-Security": "max-age=31536000; includeSubDomains",
+            "X-Content-Type-Options": "nosniff",
+        })}'"
+    )
+
+
+def _get_upload_commands(pattern: str, mime_type: str, max_age: str) -> tuple[str, str, str]:
     return (
         f"aws s3 sync dist/ {ARTIFACTS_BUCKET} --cache-control 'max-age=60' --delete --no-progress",
         f"aws s3 cp {ARTIFACTS_BUCKET} {ARTIFACTS_BUCKET} --recursive --no-progress --exclude '*' "
         f"--include {pattern} --metadata-directive REPLACE --content-type '{mime_type}' --cache-control '{max_age}'",
+        _get_metadata_command(),
```

Comment:
I think it would be important to check whether this overrides `cache-control` or not.

Reply:
I will not override the other settings.
But we have to check if it works like it is, because during the upload `x-amz-meta-` will be added automatically as a prefix...

![Bildschirmfoto 2023-11-08 um 15 02 28](https://github.com/moneymeets/action-s3-artifact/assets/38068686/0a8cc5d8-25c6-488d-908d-032a334278a0)

Reply:
So another option would be a CloudFront function, which returns the headers for the response. But we will execute the function for every response for the app... :(
Will check what options we have.

Reply:
:D @zyv Same comment and idea.

Reply:
> But we will execute the function for every response for the app... :(

Yes, but this with Lambda@Edge I think is annoying, but non-factor in terms of both costs and performance.

Reply:
Oh wow, this is cool. Do you have a link to the docs?

Reply:
> Oh wow, this is cool. Do you have a link to the docs?

https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/understanding-response-headers-policies.html#understanding-response-headers-policies-security

There are some pre-defined ones, which are not exactly what we need, but we can create our custom one.

https://us-east-1.console.aws.amazon.com/cloudfront/v4/home#/policies/responseHeaders/create

Reply:
So we will close this PR and create a new one in `moneymeets-pulumi` by settings the response headers with CloudFront. There are more security headers we could set, which can be discussed.